### PR TITLE
Cleanup + basic HA

### PR DESCRIPTION
**Pull request by Sander-agfa**
- Added basic HA Awareness,
only deploy to active HA cluster members
- Cleanup code a bit, centralise all config in dehydrated-bigip-loadconfig
- Cleanup code, centralise functions in dehydrated-bigip-common. Remove duplicate code
- Simplify installation. Removed some files. Added file dehydrated-bigip-hookscript. Here, all combinations of DNS and deployment are possible
These changes mean that script files themselves don't have to be touched anymore. Only central configfile and hookfile need to be modified. Makes it easier to update script in case of updates

### CODE REVIEW

File: dehydrated-bigip-common [hunk 1]
```diff
@@ -68,6 +69,21 @@ whenAvailable() {
...
 }
 
+#-------------------------------------------------------
+# Function: isActive()
+#   Tests to see if BIGIP is Active in HA cluster.
+#-------------------------------------------------------
+isActive() {
+  OUT=$(restCall "GET" "/tm/cm/failover-status")
+  log "isActive():`echo $OUT`"
+  if [[ $(echo $OUT | jsonq '["entries"]["https://localhost/mgmt/tm/cm/failover-status/0"]["nestedStats"]["entries"]["status"]["description"]') == ACTIVE ]]; then
```
Comment (hildebrau) [inline dehydrated-bigip-common]: shouldn't localhost be $BIGIP_DEVICE or something?
```suggestion
  if [[ $(echo $OUT | jsonq '["entries"]["https://$BIGIP_DEVICE/mgmt/tm/cm/failover-status/0"]["nestedStats"]["entries"]["status"]["description"]') == ACTIVE ]]; then
```

File: dehydrated-bigip-redeploy-traffic-certificate [hunk 2]
```diff
@@ -1,15 +1,32 @@
...
+  echo "entering unchanged_cert()"
   for BIGIP_DEVICE in ${BIGIP_DEVICE_LIST} ; do
-    ${BASEDIR}/${REDEPLOY_USING_HOOK} deploy_cert $@ ${BIGIP_DEVICE}
+	echo "check if ${BIGIP_DEVICE} is Active in cluster";
+	if isActive ; then
+	  echo "Active device, check if cert $@  is installed on ${BIGIP_DEVICE}"
+	  CERTFILE="${3}"
+	  CLIENTSSL_NAME="${1}"
+	  #Use Openssl to check the serial of the local cert, use sed to strip off "serial=" and convert to lowercase
+	  LOCALHASH=$(openssl x509 -in $CERTFILE -serial -noout |sed -E -s 's/serial=\(.*\)/\L\1/')
```
Comment (hildebrau) [inline dehydrated-bigip-redeploy-traffic-certificate]: I had to remove these escape characters to get this to parse properly on my CentOS 7 box
```suggestion
	  LOCALHASH=$(openssl x509 -in $CERTFILE -serial -noout |sed -E -s 's/serial=(.*)/\L\1/')
```

File: dehydrated-bigip-common [hunk 3]
```diff
@@ -243,6 +263,50 @@ createClientSSLProfile() {
...
+	  if [ "`echo ${CERT} | grep '"kind":"tm:sys:file:ssl-cert:ssl-certstate"'`x" != "x" ] ; then
+	    SERIAL=$(echo $CERT | jsonq '["serialNumber"]' | sed -E -s "s|:||g" )
+		log "getClientSSLProfile() SERIAL: $SERIAL"
+		echo $SERIAL
+	  else
+	    log "Cert not found or valid"
+		return 3
+	  fi
+    else
+      log "SSL cert link not found, return empty"
```
Comment (hildebrau) [inline dehydrated-bigip-common:298]: @Sander-agfa , what is supposed to happen if the certReference doesn't exist in the current cert?  It just seems to exit with return code 2 here and I'm not sure how to remedy the situation.

File: dehydrated-bigip-redeploy-traffic-certificate (hunk 2 above)
Comment (Sander-agfa) [reply dehydrated-bigip-redeploy-traffic-certificate]: This alteration does not work on by ubuntu box.
Will replace with this:

```suggestion
	  LOCALHASH=$(openssl x509 -in $CERTFILE -serial -noout |sed -e 's/serial=//' | tr '[:upper:]' '[:lower:]'
```

Comment (Sander-agfa) [reply dehydrated-bigip-redeploy-traffic-certificate]: fix with latest commit in #10 .
Uses tr in stead of sed

File: dehydrated-bigip-common (hunk 3 above)
Comment (Sander-agfa) [reply dehydrated-bigip-common:298]: @hildebrau , I should expect the function to return code 2 to the caller, and not let the script exit entirely.
I'll test this further on my dev setup, though I had this tested already.

**Review by colin-stubbs — APPROVED** [review]
looks good

**Pull request merged**